polygon/sync/service.go

```diff
@@ -47,6 +50,11 @@ func NewService(
 	p2pService := p2p.NewService(maxPeers, logger, sentryClient, statusDataProvider.GetStatusData)
 	heimdallClient := heimdall.NewHeimdallClient(heimdallUrl, logger)
 	heimdallService := heimdall.NewHeimdall(heimdallClient, logger)
+	heimdallScraper := heimdall.NewScraperTODO(
```

Comment:
In my mind this "scrapper" is an internal concept of the Heimdall Service and it should not be exposed outside of the `heimdall` package.

I would suggest to move this inside `heimdall.Heimdall` (we should probably rename this to `heimdall.Service` for consistency) and have the Heimdall service instantiate the scrapper internally in its constructor. Then here in sync we would simply just do
```
heimdallService := heimdall.NewService(heimdallClient, logger)
...
```
& in sync.Run
```
heimdallService.Run(ctx)
```

Reply:
I agree. It will be a part of HeimdallService. HeimdallService will be implemented in the next PRs. This code has `TODO` to indicate that it is temporary here.